navigation-compose/src/main/kotlin/com/freeletics/khonshu/navigation/compose/NavHost.kt

```diff
@@ -138,17 +139,20 @@ private fun InternalNavHost(
     modifier: Modifier = Modifier,
     deepLinkHandlers: Set<DeepLinkHandler> = emptySet(),
     deepLinkPrefixes: Set<DeepLinkHandler.Prefix> = emptySet(),
-    navController: NavHostController = rememberNavController(),
+    navEventNavigator: NavEventNavigator? = null,
     destinationChangedCallback: ((BaseRoute) -> Unit)? = null,
     transitionAnimations: NavHostTransitionAnimations = NavHostDefaults.transitionAnimations(),
 ) {
     val context = LocalContext.current
 
+    DisposableEffect(context, deepLinkHandlers, deepLinkPrefixes) {
```

Comment:
I think it's worth to add a comment here

```suggestion
    // should be called before rememberNavController to fill intent data with deeplinks
    DisposableEffect(context, deepLinkHandlers, deepLinkPrefixes) {
```